podman/domain/config.py

```diff
@@ -155,19 +157,128 @@ def services(self):
     @cached_property
     def active_service(self):
         """Optional[ServiceConnection]: Returns active connection."""
-
         # read the new json file format
         connection = self.attrs.get("Connection")
         if connection:
             active = connection.get("Default")
-            destinations = connection.get("Connections")
-            return ServiceConnection(active, attrs=destinations[active])
+            destinations = connection.get("Connections", {})
+            if active in destinations:
+                return ServiceConnection(active, attrs=destinations[active])
 
         # if we are here, that means there was no default in the new json file
         engine = self.attrs.get("engine")
         if engine:
             active = engine.get("active_service")
-            destinations = engine.get("service_destinations")
-            return ServiceConnection(active, attrs=destinations[active])
+            destinations = engine.get("service_destinations", {})
+            if active in destinations:
+                return ServiceConnection(active, attrs=destinations[active])
 
         return None
+
+    @cached_property
+    def network_options(self) -> Dict[str, list]:
+        """
+        Retrieves network options for all configured networks.
+
+        Returns:
+            Dict[str, list]: A dictionary where keys are network names and values are lists of options.
+        """
+        network_opts = {}
+        network_config = self.attrs.get("network", {})
+        for network_name, config in network_config.items():
+            # Assuming network options are stored as 'pasta_options', 'bridge_options', etc.
+            options_key = f"{network_name}_options"
+            if options_key in network_config:
+                network_opts[network_name] = network_config[options_key]
+        return network_opts
+
+
+def create_container_with_pasta(network_name: str, port_mapping: str, image: str, container_name: str, **kwargs):
```

Comment:
There is nothing special about pasta API wise, all the networks settings should be able to be set already via the normal API.